src/nameclaim.cpp

```diff
@@ -99,35 +106,42 @@ bool DecodeClaimScript(const CScript& scriptIn, int& op, std::vector<std::vector
             return false;
         }
     }
-    if (op == OP_UPDATE_CLAIM)
+
+    if (!scriptIn.GetOp(pc, opcode, vchParam3))
+    {
+        return false;
+    }
+    auto last_drop = OP_DROP;
+    if (opcode >= 0 && opcode <= OP_PUSHDATA4 && op != OP_CLAIM_NAME)
```

Comment:
It does not look correct or we miss OP_SUPPORT_CLAIM && !allowSupportMetadata it's same as OP_CLAIM_NAME ?

Reply:
```
switch (op) {
    case OP_SUPPORT_CLAIM:
        if (allowSupportMetadata)
        // fallthrough
    case OP_UPDATE_CLAIM: {
        if (opcode < 0 || opcode > OP_PUSHDATA4)
            return false;
        if (!scriptIn.GetOp(pc, opcode))
            return false;
    } // fallthrough
    case OP_CLAIM_NAME:
        if (opcode != OP_2DROP)
           return false;
    break;
}
```

Reply:
I don't understand your concern here. The logic is "if not a claim op and have another data chunk", that's fine -- that's what we want. That third data is required on the update op but not required on the support op.

Reply:
I was afraid on op == OP_SUPPORT_CLAIM and allowSupportMetadata == false we can enter in by the condition but opcode is OP_2DROP so it's OK.